spec/models/manageiq/providers/kubernetes/container_manager/event_catcher_mixin_spec.rb

```diff
@@ -0,0 +1,267 @@
+          :uid                       => '7599d451-4c1c-11e6-89dd-525400c7c086',
+          :container_replicator_name => 'mysql-1',
+          :container_namespace       => 'proj',
+          :event_type                => 'REPLICATOR_SUCCESSFULCREATE'
+        }
+        event = RecursiveOpenStruct.new(:object => kubernetes_event)
+        expect(dummy_class.new.extract_event_data(event)).to eq(expected_data)
+      end
+    end
+
+    context 'given node event' do
+      let(:kubernetes_event) do
+        {
+          'metadata'       => {
+            'name'              => 'vm-test-03.example.com.146481d4a19edff4',
+            'namespace'         => 'default',
+            'selfLink'          => '/api/v1/namespaces/default/events/vm-test-03.example.com.146481d4a19edff4',
+            'uid'               => 'a4b92ae1-5251-11e6-8564-525400c7c086',
+            'resourceVersion'   => '1356588',
+            'creationTimestamp' => '2016-07-25T10:22:09Z',
+            'deletionTimestamp' => '2016-07-25T12:22:09Z'
+          },
+          'involvedObject' => {
+            'kind' => 'Node',
+            'name' => 'vm-test-03.example.com',
+            # Actually captured 'vm-test-03.example.com' but is being fixed in kubernetes.
+            # https://github.com/kubernetes/kubernetes/issues/29289
+            'uid'  => 'd30a880d-dfa7-11e5-af89-525400c7c086'
+          },
+          'reason'         => 'Rebooted',
+          'message'        => 'Node vm-test-03.example.com has been rebooted, boot id: '\
+                              'c75d2b66-6d5b-49e0-b906-1d8abaf3e73b',
+          'source'         => {
+            'component' => 'kubelet',
+            'host'      => 'vm-test-03.example.com'
+          },
+          'firstTimestamp' => '2016-07-25T10:22:08Z',
+          'lastTimestamp'  => '2016-07-25T10:22:08Z',
+          'count'          => 1,
+          'type'           => 'Warning'
+        }
+      end
+
+      let(:expected_data) do
+        {
+          :timestamp           => '2016-07-25T10:22:08Z',
+          :kind                => 'Node',
+          :name                => 'vm-test-03.example.com',
+          :namespace           => nil,
+          :reason              => 'Rebooted',
+          :message             => 'Node vm-test-03.example.com has been rebooted, boot id: '\
+                                  'c75d2b66-6d5b-49e0-b906-1d8abaf3e73b',
+          :uid                 => 'd30a880d-dfa7-11e5-af89-525400c7c086',
+          :container_node_name => 'vm-test-03.example.com',
+          :event_type          => 'NODE_REBOOTED'
+        }
+      end
+
+      it 'given good uid extracts NODE_REBOOTED event data' do
+        event = RecursiveOpenStruct.new(:object => kubernetes_event)
+        expect(dummy_class.new.extract_event_data(event)).to eq(expected_data)
+      end
+
+      context 'given useless/missing uid' do
+        # We've seen events with both missing uid and uid == name.
+        let(:bad_uid_event) do
```

Comment:
@cben please add a comment with a link to the kubernetes issue saying we should remove this test once the issue is fixed